sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/common/FileBasedIOITHelper.java

```diff
@@ -65,6 +71,36 @@ public void processElement(ProcessContext c) {
     }
   }
 
+  public static void publishTestMetrics(
+      PipelineResult result,
+      String bigQueryTable,
+      String bigQueryDataset,
+      String namespace,
+      String writeStartName,
+      String writeEndName,
+      String readStartName,
+      String readEndName) {
+    if (bigQueryTable != null && bigQueryDataset != null) {
+      MetricsReader reader = new MetricsReader(result, namespace);
+      long writeStart = reader.getStartTimeMetric(writeStartName);
+      long writeEnd = reader.getEndTimeMetric(writeEndName);
+      long readStart = reader.getStartTimeMetric(readStartName);
+      long readEnd = reader.getEndTimeMetric(readEndName);
+      double writeTime = (writeEnd - writeStart) / 1000.0;
+      double readTime = (readEnd - readStart) / 1000.0;
+
+      String uuid = UUID.randomUUID().toString();
```

Comment:
Please use the same testID for all NamedTestResuts in this test.

*Test id makes selecting all metrics from a particular test run easier in BigQuery. Because of that, the testID should be the same for all metrics in a test invocation.

Reply:
Ok, done